### PR TITLE
Making the package manager detect and report errors for source paths …

### PR DESCRIPTION
**Pull request by harshgMSFT**
…which do not exist. Fixs #594.

### CODE REVIEW

**Comment by harshgMSFT** [comment]
@yishaigalatzer @pranavkm @danliu

File: src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs [hunk 1]
```diff
@@ -43,6 +44,13 @@ public UISearchResourceV2(IPackageRepository repo)
         {
             return await Task.Run(() =>
             {
+                // Check if source is available.
```
Comment (yishaigalatzer) [inline src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs:47]: This comment should say: Check if the source is valid

File: src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs [hunk 2]
```diff
@@ -43,6 +44,13 @@ public UISearchResourceV2(IPackageRepository repo)
...
+                if (!IsHttpSource(V2Client.Source) && !IsLocalOrUNC(V2Client.Source))
+                {
+                    throw new InvalidOperationException(
```
Comment (yishaigalatzer) [inline src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs:50]: I assume you verified we don't crash here, and we show a proper error message

**Comment by yishaigalatzer** [comment]
:shipit:

File: src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs (hunk 2 above)
Comment (harshgMSFT) [inline src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs:50]: yes and yes.

File: src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs (hunk 1 above)
Comment (harshgMSFT) [inline src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs:47]: the source might be valid, but not available (network issues etc), .. if you think valid is better ill change that.

Comment (yishaigalatzer) [inline src/NuGet.Protocol.VisualStudio/UISearchResourceV2.cs:47]: The comment reads as applying only to the if. You might want to clarify.

I'm ok either way 

-----Original Message-----

**Pull request closed without merging**